modules/integration/src/main/java/webindex/integration/DevServer.java

```diff
@@ -0,0 +1,169 @@
+    cluster.start();
+
+    FluoConfiguration config = new FluoConfiguration();
+    config.setMiniStartAccumulo(false);
+    config.setApplicationName("webindex-dev");
+    config.setAccumuloInstance(cluster.getInstanceName());
+    config.setAccumuloUser("root");
+    config.setAccumuloPassword("secret");
+    config.setInstanceZookeepers(cluster.getZooKeepers() + "/fluo");
+    config.setAccumuloZookeepers(cluster.getZooKeepers());
+    config.setAccumuloTable("webindex");
+    config.setWorkerThreads(5);
+
+    String exportTable = "webindex_search";
+
+    log.info("Initializing Accumulo & Fluo");
+    IndexEnv env = new IndexEnv(config, exportTable, "/tmp", TEST_SPLITS, TEST_SPLITS);
+    env.initAccumuloIndexTable();
+    env.configureApplication(config);
+
+    FluoFactory.newAdmin(config).initialize(
+        new FluoAdmin.InitializationOptions().setClearTable(true).setClearZookeeper(true));
+
+    env.setFluoTableSplits();
+
+    log.info("Starting web server");
+    client = new IndexClient(exportTable, cluster.getConnector("root", "secret"));
+    webServer.start(client, webPort, templatePath);
+
+    log.info("Loading data from {}", dataPath);
+    Gson gson = new Gson();
+    try (MiniFluo miniFluo = FluoFactory.newMiniFluo(config);
+        FluoClient client = FluoFactory.newClient(miniFluo.getClientConfiguration())) {
+
+      try (LoaderExecutor le = client.newLoaderExecutor()) {
+
+        Files
+            .lines(dataPath)
+            .map(json -> Page.fromJson(gson, json))
+            .forEach(
+                page -> {
+                  log.debug("Loading page {} with {} links", page.getUrl(), page.getOutboundLinks()
+                      .size());
+                  le.execute(PageLoader.updatePage(page));
+                });
+      }
+
+      log.info("Finished loading data. Waiting for observers to finish...");
+      miniFluo.waitForObservers();
+      log.info("Observers finished");
+    }
+
+    started.set(true);
+  }
+
+  public void stop() throws Exception {
+    webServer.stop();
+    cluster.stop();
+  }
+
+  public static void main(String[] args) throws Exception {
+    String dataLocation = "data/1K-pages.txt";
+    String templateLocation = "modules/ui/src/main/resources/spark/template/freemarker";
+    if (args.length == 2) {
+      dataLocation = args[0];
+      templateLocation = args[1];
+    }
+    log.info("Looking for data at {}", dataLocation);
+
+    Path dataPath = Paths.get(dataLocation);
+    if (Files.notExists(dataPath)) {
+      log.info("Generating sample data at {} for dev server", dataPath);
+      SampleData.generate(dataPath, 1000);
+    }
+
+    Path templatePath = Paths.get(templateLocation);
+    if (Files.notExists(templatePath)) {
+      log.info("Template location {} does not exits", templateLocation);
+      throw new IllegalArgumentException("Template location does not exist");
+    }
+
+    DevServer devServer = new DevServer(dataPath, Paths.get("/tmp/"), 4567, templatePath);
```

Comment:
Could used Java 8 [new function to create temp dir](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#createTempDirectory-java.lang.String-java.nio.file.attribute.FileAttribute...-)